### PR TITLE
Extra test cases Added to word_extraction_test

### PR DESCRIPTION
**Pull request by abenezer-gelaw**
This PR contains extra test cases to test edge cases of the vocabulary tool returning empty vocab or null vocab artifacts. This PR completes the work done on completing the word extraction logic.

### CODE REVIEW

**Review by shrutiparabgoogle — COMMENTED** [review]
First look NIT: 

Can we keep only two folders in the `consistency/tool` folder :
1. `clustering` -> should contain extraction and clustering logic, since they are actually dependent, (Let's not make a new extraction folder)
2. `comparison` -> which will contain the comparison logic.

We are essentially building only two tools overall, cluster and compare.

File: consistency/tool/clustering/word_extraction_test.json [hunk 1]
```diff
@@ -18,5 +18,55 @@
       "parameters":  [{"word": "ab", "count":2}, {"word": "bc", "count":1},{"word": "cd", "count":1} ]
    }
     
-  ]
+  ], 
+
+  "test2": [
```
Comment (shrutiparabgoogle) [inline consistency/tool/clustering/word_extraction_test.json:23]: Can we make the test_ids more descriptive? Instead of "test1", "test2", ... Something like "simple-test", "empty-operations", etc.

File: consistency/tool/clustering/word_extraction_test.json [hunk 2]
```diff
@@ -18,5 +18,55 @@
...
+         "name":"artifact1", 
+         "schemas":     [],
+         "properties" : [],
+         "operations" : [],
+         "parameters":  []
+      }
+
+     
+   ],
+   "test5": [
```
Comment (shrutiparabgoogle) [inline consistency/tool/clustering/word_extraction_test.json:59]: Can we add a test case where the `get_vocabs` doesn't return any vocabulary artifacts, essentially an empty list. 

Also, what if `get_vocabs` returns an error? Because of unable to connect to the registry?

File: consistency/tool/clustering/word_extraction.py [hunk 3]
```diff
@@ -7,32 +7,45 @@ def __init__(self, stub):
...
-          contents = stub.GetArtifactContents(
-              registry_service_pb2.GetArtifactContentsRequest(
-                  name=artifact.name,
-              )
-          )
-          vocab = vocabulary_pb2.Vocabulary()
-          vocabs.append(vocab.ParseFromString(contents.data))
-
-      return vocabs
+        stub = self.stub
```
Comment (shrutiparabgoogle) [inline consistency/tool/clustering/word_extraction.py:10]: NIT: This line is not required I think, can you `self.stub` directly.

File: consistency/tool/clustering/word_extraction.py [hunk 4]
```diff
@@ -7,32 +7,45 @@ def __init__(self, stub):
...
+        try:
+            response = stub.ListArtifacts(
+                registry_service_pb2.ListArtifactsRequest(
+                    parent="projects/-/locations/global/apis/-/versions/-/specs/-",
+                    filter="name.contains(\"vocabulary\")"
+                ))
+        except grpc.RpcError as rpc_error:
+            print(f"Received RPC error: code={rpc_error.code()} message={rpc_error.details()}")
```
Comment (shrutiparabgoogle) [inline consistency/tool/clustering/word_extraction.py:18]: Can we make this error message more descriptive like "Failed to fetch vocabulary artifacts, RPC error: ..."

File: consistency/tool/clustering/word_extraction.py [hunk 5]
```diff
@@ -7,32 +7,45 @@ def __init__(self, stub):
...
+                registry_service_pb2.GetArtifactContentsRequest(
+                    name=artifact.name
+                )
+            )
+            vocab = vocabulary_pb2.Vocabulary()
+
+            try:
+                vocabs.append(vocab.ParseFromString(contents.data))
+            except Exception as e:
+                print(e, " Parsing failed ", failures, "times!")
```
Comment (shrutiparabgoogle) [inline consistency/tool/clustering/word_extraction.py:34]: Instead of number of failures, can we specify the name of the artifact which failed in parsing.

File: consistency/tool/clustering/word_extraction_test.json [hunk 6]
```diff
@@ -18,5 +18,58 @@
...
+         "name":"artifact1", 
+         "schemas":     null,
+         "properties" : null,
+         "operations" : null,
+         "parameters":  null
+      }
+
+     
+   ], 
+   "no-return-values": [
```
Comment (shrutiparabgoogle) [inline consistency/tool/clustering/word_extraction_test.json:71]: NIT: Can combine lines 71-73 into `"no-return-values": []`

**Review by shrutiparabgoogle — APPROVED** [review]
Approved with minor NIT comments.
